pkg/globalconfig/global_config.go

```diff
@@ -475,11 +491,18 @@ func ReadProjectList() error {
 			return nil
 		}
 		if os.IsNotExist(err) {
-			// write an empty file
-			err := os.WriteFile(GetProjectListPath(), make([]byte, 0), 0644)
-			if err != nil {
+			// If someone upgrades from an earlier version, the global config may hold the project list.
+			if ddevLegacyGlobalConfig.ProjectList != nil && len(ddevLegacyGlobalConfig.ProjectList) > 0 {
+				DdevProjectList = ddevLegacyGlobalConfig.ProjectList
+				err := WriteProjectList(DdevProjectList)
+				// Whether there's an error or nil here we want to return
 				return err
 			}
```

Comment:
We only want to check legacy config if the file doesn't exist. If the file exists, it's the source of truth even if it's empty.